modules/jolt_physics/shapes/jolt_concave_polygon_shape_3d.cpp

```diff
@@ -29,13 +29,25 @@
 /**************************************************************************/
 
 #include "jolt_concave_polygon_shape_3d.h"
+#include "jolt_custom_material_override_shape.h"
 
 #include "../jolt_project_settings.h"
 #include "../misc/jolt_type_conversions.h"
 
 #include "Jolt/Physics/Collision/Shape/MeshShape.h"
 
-JPH::ShapeRefC JoltConcavePolygonShape3D::_build() const {
+void JoltConcavePolygonShape3D::_set_material(JPH::RefConst<JoltPhysicsMaterial> &material) {
+	if (!jolt_ref) {
+		return;
+	}
+	jolt_ref_mutex.lock();
+	JoltCustomMaterialOverrideShape *ref = static_cast<JoltCustomMaterialOverrideShape *>(jolt_ref.GetPtr());
```

Comment:
I don't think you need `JoltCustomMaterialOverrideShape`. You can use:

```
PhysicsMaterialRefC array[] = { mat };
MeshShape::RestoreMaterialState(array, 1);
```

instead.